### DISCUSSION TITLE
Archive = false still creates version history

### DISCUSSION THREAD

**Discussion by vedeja** [structurizr/cli#144] [Q&A] [answered]
### Question

I am pushing using the below command. Since archive is set to false, I'm expecting the current version to be overwritten and no history remain. But I can see the number of workspace version files growing for each push, and the version dropdown in the workspace ui lists them all. What is wrong here, my expectation, my command or the cli?

`structurizr/cli push -url https://xxx/api -id 1 -key mykey -secret mysecret -workspace workspace.json -merge false -archive false`

**Comment by simonbrowndotje** [comment, marked as answer]
The `archive` flag prevents a local copy of the workspace being stored as a backup ... the cloud service/on-premises installation will still store workspace versions, and this is what you're seeing in the UI.

**Reply by vedeja** [reply]
Thanks. After reading your response and the docs at https://docs.structurizr.com/cli/push, it makes sense. That said, do you think there is a benefit in making the cloud service/on-premises installation willing to stop versioning by overwriting (the way I described)? The reasoning is that we already version control the dsl files and can go back to an older version anyway, should we have to.

**Reply by simonbrowndotje** [reply]
> do you think there is a benefit in making the cloud service/on-premises installation willing to stop versioning by overwriting

You can set `structurizr.maxWorkspaceVersions` for the on-premises installation to `0`, which should prevent versioning; see https://docs.structurizr.com/onpremises/configuration#structurizrproperties for more details.

> we already version control the dsl files and can go back to an older version anyway,

If you're only using automatic layout, yes, that could work.

**Reply by vedeja** [reply]
Thanks. I tried `structurizr.maxWorkspaceVersions=0` but it must be doing something else than what I expect, because versioning continues.
